src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -486,7 +486,10 @@ private boolean allowDuplicateChannelName(Channel channel) {
     private static String generateChannelNameFromTitle(@NotNull String prefix, Optional<String> title) {
         String channelName = prefix + title.orElse("");
         // [^a-z0-9]+ matches all occurrences of single or consecutive characters that are no digits and letters
-        channelName = channelName.toLowerCase().replaceAll("[^a-z0-9]+", "-");
+        String specialCharacters = "[^a-z0-9]+";
```

Comment:
This could be `private static final String`

Reply:
I think I will leave it like this, since the regex ist not relevant to the whole class but only this small helper method.